nestkernel/event.h

```diff
@@ -827,7 +843,31 @@ class GapJEvent : public SecondaryEvent
   static void
   set_syn_id( const synindex synid )
   {
-    synid_ = synid;
+    if ( synid_ == invalid_synindex )
+      synid_ = synid;
+
+    synid_vector_.push_back( synid );
+  }
+
+  void
+  add_syn_id( const synindex synid )
+  {
+    synid_vector_.push_back( synid );
+  }
+
+  bool
+  has_syn_id( const synindex synid ) const
+  {
+    bool has_syn_id = false;
+    for ( unsigned int i = 0; i < synid_vector_.size(); i++ )
+    {
+      if ( synid_vector_[ i ] == synid )
+      {
+        has_syn_id = true;
+        break;
+      }
+    }
+    return has_syn_id;
```

Comment:
You can reduce this method to (assuming here the renaming suggested above):
```
{
  return supported_syn_ids_.find( synid ) != supported_syn_ids_.end();
}
```
This gives you the full speed of the STL implementation and will work for any container type.